tests/snmp/test_snmp_queue_counters.py

```diff
@@ -48,10 +48,28 @@ def test_snmp_queue_counters(duthosts,
         = "docker exec snmp snmpwalk -v2c -c {} {} {}".format(
             creds_all_duts[duthost.hostname]['snmp_rocommunity'], hostip,
             Ethernet0_queue_cntrs_oid)
+    # Generate sonic-cfggen commands for multi-asic and single-asic duts
+    nslist = duthost.get_asic_namespace_list()
+    if duthost.sonichost.is_multi_asic and nslist is not None:
+        cmd = "sonic-cfggen -n asic0 -d --print-data > {}".format(ORIG_CFG_DB)
+    else:
+        cmd = "sonic-cfggen -d --print-data > {}".format(ORIG_CFG_DB)
 
-    duthost.shell("sonic-cfggen -d --print-data > {}".format(ORIG_CFG_DB))
+    duthost.shell(cmd)
     data = json.loads(duthost.shell("cat {}".format(ORIG_CFG_DB),
                                     verbose=False)['stdout'])
+    buffer_queue_to_del = None
+    # Get appropriate buffer queue value to delete for Ethernet0|3-4 in case of multi-asic
+    if duthost.sonichost.is_multi_asic:
+        buffer_queues = list(data['BUFFER_QUEUE'].keys())
+        iface_to_check = buffer_queues[0].split('|')[0]
+        iface_buffer_queues = [bq for bq in buffer_queues if any(val in iface_to_check for val in bq.split('|'))]
+        for queue in iface_buffer_queues:
+            if 'asic0' in queue and queue.split('|')[-1] == '3-4':
+                buffer_queue_to_del = queue
+                break
+    else:
+        buffer_queue_to_del = "Ethernet0|3-4"
```

Comment:
Does the test always expect to have queue configuration for Ethernet0 and delete that?

Reply:
Ethernet0 was chosen as a random arbitrary port when this test was added initially by Aryeh. Similarly, the current changes for multi-asic, Ethernet0 is being chosen for asic0 and Ethernet144 is being chosen for asic1.

Reply:
Can we find the asic interface dynamically?
Not all HwSKU might have Ethernet144, for example the multi-asic VS does not have Ethernet144 interface https://github.com/sonic-net/sonic-buildimage/blob/master/device/virtual/x86_64-kvm_x86_64_4_asic-r0/msft_four_asic_vs/1/port_config.ini